core/src/main/scala/org/apache/spark/rdd/RDDOperationScope.scala

```diff
@@ -95,17 +95,17 @@ private[spark] object RDDOperationScope extends Logging {
   private[spark] def withScope[T](
       sc: SparkContext,
       allowNesting: Boolean = false)(body: => T): T = {
-    val stackTrace = Thread.currentThread.getStackTrace().tail // ignore "Thread#getStackTrace"
-    val ourMethodName = stackTrace(1).getMethodName // i.e. withScope
-    // Climb upwards to find the first method that's called something different
-    val callerMethodName = stackTrace
-      .find(_.getMethodName != ourMethodName)
+
+    val callerMethodName = Thread.currentThread.getStackTrace()
+      .dropWhile(! _.getMethodName().equals("withScope"))
```

Comment:
The original code captures the stack trace within this method - at line 99, this will always contain `withScope` as the code comment suggests (unless this method name changes). 
The new code is dropping elements in the stack trace until we see `withScope` and follows the same logic henceforth.